src/commando/core.py

```diff
@@ -1,2 +1,79 @@
-def hello_world():
-    return "Hello, world!"
+import subprocess
+import tempfile
+from functools import partial
+from subprocess import PIPE, STDOUT, CalledProcessError
+
+
+def _pprint(repo, message):
+    print(f"{repo}> {message}")
+
+
+def _run(cwd, results, command, **kwargs):
+    result = subprocess.run(  # pylint:disable=subprocess-run-check
+        command, cwd=cwd, stdout=PIPE, stderr=STDOUT, text=True, **kwargs
+    )
+    results.append(result)
+    result.check_returncode()
+    return result
+
+
+def _process_repo(
+    repo, command, commit_message, branch, pr_title, pr_body
+):  # pylint:disable=too-many-arguments
+
+    # Collect the results of all commands run for this repo.
+    results = []
+
+    pprint = partial(_pprint, repo)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        run = partial(_run, tmpdirname, results)
+
+        try:
+            pprint("cloning")
+            run(["gh", "repo", "clone", repo, tmpdirname, "--", "--quiet"])
+
+            pprint(command)
+            run(command, shell=True)
+
+            if run(["git", "status", "--porcelain"]).stdout:
+                pprint("committing changes")
+                run(["git", "switch", "--force-create", branch, "origin/main"])
```

Comment:
Some `git switch` commands will create a remote tracking branch if there's a remote with a branch with the same name. Need to make sure that this particular form of `git switch` doesn't do that because we don't want it to. I think we need to add `--no-guess` to be safe. We may want to stick in `--no-track` just for good measure too (I don't think upstream tracking information would do any harm but we don't want it).